tests/snappi_tests/multidut/ecn/files/multidut_helper.py

```diff
@@ -0,0 +1,181 @@
+    generate_pause_flows, run_traffic                                       # noqa: F401
+
+logger = logging.getLogger(__name__)
+
+EXP_DURATION_SEC = 1
+DATA_START_DELAY_SEC = 0.1
+SNAPPI_POLL_DELAY_SEC = 2
+PAUSE_FLOW_NAME = 'Pause Storm'
+DATA_FLOW_NAME = 'Data Flow'
+
+
+def run_ecn_test(api,
+                 testbed_config,
+                 port_config_list,
+                 conn_data,
+                 fanout_data,
+                 dut_port,
+                 lossless_prio,
+                 prio_dscp_map,
+                 iters,
+                 snappi_extra_params=None):
+    """
+    Run a ECN test
+
+    Args:
+        api (obj): SNAPPI session
+        testbed_config (obj): testbed L1/L2/L3 configuration
+        port_config_list (list): list of port configuration
+        conn_data (dict): the dictionary returned by conn_graph_fact.
+        fanout_data (dict): the dictionary returned by fanout_graph_fact.
+        dut_port (str): DUT port to test
+        lossless_prio (int): lossless priority
+        prio_dscp_map (dict): Priority vs. DSCP map (key = priority).
+
+    Returns:
+        Return captured IP packets (list of list)
+    """
+
+    if snappi_extra_params is None:
+        snappi_extra_params = SnappiTestParams()
+
+    duthost1 = snappi_extra_params.multi_dut_params.duthost1
+    rx_port = snappi_extra_params.multi_dut_params.multi_dut_ports[0]
+    duthost2 = snappi_extra_params.multi_dut_params.duthost2
+    tx_port = snappi_extra_params.multi_dut_params.multi_dut_ports[1]
+    iters = snappi_extra_params.test_iterations
+
+    pytest_assert(testbed_config is not None, 'Failed to get L2/3 testbed config')
+
+    logger.info("Stopping PFC watchdog")
+    stop_pfcwd(duthost1, rx_port['asic_value'])
+    stop_pfcwd(duthost2, tx_port['asic_value'])
+    logger.info("Disabling packet aging if necessary")
+    disable_packet_aging(duthost1)
+    disable_packet_aging(duthost2)
+
+    # Configure WRED/ECN thresholds
+    logger.info("Configuring WRED and ECN thresholds")
+    config_result = config_wred(host_ans=duthost1,
+                                kmin=snappi_extra_params.ecn_params["kmin"],
+                                kmax=snappi_extra_params.ecn_params["kmax"],
+                                pmax=snappi_extra_params.ecn_params["pmax"])
+    pytest_assert(config_result is True, 'Failed to configure WRED/ECN at the DUT')
+    config_result = config_wred(host_ans=duthost2,
+                                kmin=snappi_extra_params.ecn_params["kmin"],
+                                kmax=snappi_extra_params.ecn_params["kmax"],
+                                pmax=snappi_extra_params.ecn_params["pmax"])
+    pytest_assert(config_result is True, 'Failed to configure WRED/ECN at the DUT')
+
+    # Enable ECN marking
+    logger.info("Enabling ECN markings")
+    enable_ecn(host_ans=duthost1, prio=lossless_prio)
+    enable_ecn(host_ans=duthost2, prio=lossless_prio)
```

Comment:
can we do a pytest assert on this ^ in case the command fails?

Reply:
Fixed